mgr/cephadm: Add virtual_ip as a param for mgmt-gateway apply cmd (#61829)
ceph/ceph: doc/cephadm/services/mgmt-gateway.rst

==========================================

In addition to providing high availability for the underlying backend services, the mgmt-gateway
service itself can be configured for high availability, ensuring that the entire system remains resilient

> even if certain core components for the service fail including the `mgmt-gateway`.

Reviewer: NIT
```suggestion
even if certain core components for the service fail, including the `mgmt-gateway` itself.
```

Author: fixed

In addition to providing high availability for the underlying backend services, the mgmt-gateway
service itself can be configured for high availability, ensuring that the system remains resilient

> even if certain core components for the service fail, including the `mgmt-gateway` itself.

Reviewer 2: Markdown rendering varies, but with Ceph docs I usually use double backticks.  Single backticks *might* be rendered as italics.  Let's see how GH renders, which might be different than Sphinx.

`single`
``double``

GH renders both of those the same, but Sphinx might.  I suggest just using double as best practice.

Reviewer 3: I shared these links in the orch standup today, but I'll share them again here in case someone stumbles across this in the future:

https://stackoverflow.com/a/57302493
https://docutils.sourceforge.io/docs/user/rst/quickref.html#inline-markup
https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#inline-markup

IOW, if you are writing reStructuredText, use double backticks for code, command lines, etc. Do not use (unadorned) single backticks unless you know what you are doing with them (making hyperlinks, references, etc). You should not use them just to make text italic.

Author: thanks @phlogistonjohn

directly accessing the `mgmt-gateway` internal endpoint, services use the virtual IP specified in the spec.
This ensures that the high availability mechanism for `mgmt-gateway` is transparent to other services.

> The simplest and recommended way to deploy the mgmt-gateway in high availability mode is by using labels. To

Reviewer 2: is `mgmt-gateway` the actual name of the thing?  If so this instance maybe should be in backticks like the one on the next line.  We are admittedly not consistent about this throughout the docs, but one day we should make a pass through to harmonize.

Author: `mgmt-gateway` is the name of the management gateway cephadm service

> run ``mgmt-gateway`` in HA mode, user can use either cephadm command line as in the following example:

Reviewer: ```suggestion
run the ``mgmt-gateway`` in HA mode users can either use the cephadm command line as follows:
```

Author: fixed

      enable_auth: true
      virtual_ip: 192.168.100.220

> In addition, the user must configure an `Ingress` service to provide virtual IP functionality for

Author: fixed

> the `mgmt-gateway`. Below is a configuration example for the ingress service:

Reviewer: NIT, I think what this is an example of is self-explanatory given the previous sentence and what's actually in the example
```suggestion
the `mgmt-gateway`. For example:
```

Author: fixed

The number of deployed instances is determined by the number of hosts with the mgmt label.
The ingress is configured in `keepalive_only` mode, with labels ensuring that any changes to
the `mgmt-gateway` daemons are replicated to the corresponding keepalived instances.

> .. note::

Reviewer: We might want to even make this a warning instead of a note. I think this is something that would be pretty easy for users to miss

Author: yeah, I'll see if we can detect that automatically :+1: (would be a separate PR anyway)

Reviewer 2: Also suggest double backticks around keepalived, and on line 102 above, keepalive_only.
